packages/vite/src/node/plugins/optimizedDeps.ts

```diff
@@ -96,7 +96,7 @@ export function optimizedDepsBuildPlugin(config: ResolvedConfig): Plugin {
 
       if (depsOptimizer.isOptimizedDepFile(id)) {
         return id
-      } else {
+      } else if (!options.ssr) {
```

Comment:
I don't understand why we do it specifically for SSR? I ran the repro and the build seems to get stuck for me. I wonder if this could also be fixed by https://github.com/vitejs/vite/pull/13127? (I forgot about that PR but I think we could also merge that too)

Reply:
My wires crossed with SSR dev here, where we only optimize what is explicitly defined. Closing this PR. I need to finish the delayLoading feat for rollup. Closing this PR.